projects/backend/image.ts

```diff
@@ -1,35 +1,51 @@
 import { createHash } from 'crypto'
-import { ImageSize } from './common'
+import { ImageSize, Image } from './common'
+import Vibrant from 'node-vibrant'
+
 const salt = process.env.IMAGE_SALT
 
-export const getImageSlug = (url: string) => {
+export const getImageFromURL = (url: string): Image => {
     const parsed = new URL(url)
-    const path = parsed.pathname
+    const path = parsed.pathname.slice(1) //remove leading slash
     const host = parsed.hostname
     const hostparts = host.split('.')
     const source = hostparts[0]
-    return source + path
+    return { source, path }
 }
 
 const getSignature = (path: string) => {
-    console.log(`${salt}${path}`)
     return createHash('md5')
         .update(`${salt}/${path}`)
         .digest('hex')
 }
 
-const getSize = (size: ImageSize) => {
-    switch (size) {
-        case 'small':
-            return 500
-        case 'notsmall':
-            return 2000
-    }
+const sizes: { [k in ImageSize]: number } = {
+    phone: 1000,
+    tablet: 1000,
+    sample: 200,
 }
 
-export const getRedirect = (source: string, size: ImageSize, path: string) => {
-    const newPath = `${path}?q=85&w=${getSize(size)}` //&dpr=2&width=500&auto=format`
-    return `https://i.guim.co.uk/img/${source}/${newPath}&s=${getSignature(
-        newPath,
-    )}`
+export const getImageURL = (image: Image, size: ImageSize) => {
+    const newPath = `${image.path}?q=25&dpr=2&w=${sizes[size]}`
```

Comment:
hey, this is the correct value assuming that the app will always support webp images. Otherwise I would set it to 45, where it will then be [rewritten to 25 if the device supports webp](https://github.com/guardian/fastly-image-service/blob/master/fastly-io_guim_co_uk/src/main/resources/varnish/main.vcl#L336)

As a first pass assuming dpr 2 seems reasonable since this is for people with lots of money on mobile devices, we might want to vary it in the future like we do on dotcom